### PR TITLE
Map `model_system` to `outputs`

### PR DESCRIPTION
**Pull request by ndaelman-hu**
As shown here, `model_method` or `model_system` references in `outputs` are only populated when there is a single, clear section.
This is now extended to cover 1-1 mapping with `model_system`.

Screenshot from 2025-05-23 14-46-33

### CODE REVIEW

File: src/nomad_simulations/schema_packages/outputs.py [hunk 1]
```diff
@@ -163,8 +163,13 @@ def set_model_system_ref(self) -> Optional[ModelSystem]:
...
         if self.m_parent is not None:
             model_systems = self.m_parent.model_system
-            if model_systems is not None and len(model_systems) == 1:
-                return model_systems[-1]
+            outputs = self.m_parent.outputs
+            if (
+                isinstance(model_systems, list)
+                and isinstance(outputs, list)
+                and len(model_systems) == len(outputs)
+            ):
```
Comment (ndaelman-hu) [inline src/nomad_simulations/schema_packages/outputs.py]: Note: this version is more restrictive (will fail with arrays, for example), but also more safe. Since `len` is a method of `list`, there's no reliance on duck typing here.

**Comment by ndaelman-hu** [comment]
@JFRudzinski If you want, we can apply this logic to the workflows too.

**Review by EBB2675 — APPROVED** [review]
Thank you very much!

Will you be adding the unit tests (e.g. matching lengths vs. mismatched lengths) as well?

**Comment by ndaelman-hu** [comment]
> Thank you very much!
> 
> Will you be adding the unit tests (e.g. matching lengths vs. mismatched lengths) as well?

Fair question. Let's see, are there any edge cases to cover? 3 cases come to mind:

1. other indexable types like `np.array` not being picked up on. Normally, `nomad` should handle the conversion of repeating subsections to `list`.
2. `m_parent_index` having a non-numerical or abnormal value. The stand-out case is `-1` for non-repeating subsections. This is also on `nomad` to enforce.
3. `IndexError` due to the lists being empty.

I patched point 3, thank you for making me re-evaluate it. Points 1 and 2 would be due to failures on `nomad`'s side.
I can check that these cases lead to failure, but recovery from them is beyond our purview.

**Comment by ndaelman-hu** [comment]
@EBB2675 Have extended the testing. Will merge, but retain the branch for a day, in case I get further feedback.
